system/jlib/jtime.cpp

```diff
@@ -117,24 +117,33 @@ time_t timelocal(struct tm * local)
 
 #endif //__GNUC__
 
-static unsigned readDigits(char const * & str, unsigned numDigits)
+static unsigned readDigits(char const *startstr, char const * & str, unsigned numDigits)
 {
     unsigned ret = 0;
     while(numDigits--)
     {
         char c = *str++;
         if(!isdigit(c))
-            throwError1(JLIBERR_BadlyFormedDateTime, str);
+            throwError1(JLIBERR_BadlyFormedDateTime, startstr);
         ret  = ret * 10 + (c - '0');
     }
     return ret;
 }
 
-static void checkChar(char const * & str, char required)
+static void checkChar(char const *startstr, char const * & str, char required)
 {
     char c = *str++;
     if(c != required)
-        throwError1(JLIBERR_BadlyFormedDateTime, str);
+        throwError1(JLIBERR_BadlyFormedDateTime, startstr);
+}
+
+static void checkChars(char const *startstr, char const * & str, char const * any)
+{
+    char c = *str++;
+    while (*any && c != *any)
+        any++;
+    if(c != *any)
```

Comment:
As it stands checkChars will succeed if c == '\0', regardless of contents of 'any'. I assume it shouldn't.
Could code as:
```
    while (*any)
    {
        if ((c == *any++)
            return;
    }
    throwError1(JLIBERR_BadlyFormedDateTime, startstr);
```